### PR TITLE
Add V0s to miniaod

### PR DESCRIPTION
**Pull request by arizzi**
Add k0 and lambdas from generalV0Candidates producer.
V0s are converted to VertexCompositePtrCandidate and rekeyed to point to packedPFcandidate.
In addition tracks from V0s are white listed 

Size increase is 150bytes/event on ttbar
http://arizzi.web.cern.ch/arizzi/miniaod/plusK0.html#recoVertexCompositePtrCandidates_slimmedKshortVertices__PAT

this PR is on top of most recent version of #18092

### CODE REVIEW

**Comment by perrotta** [comment]
please test

**Comment by alberto-sanchez** [comment]
@arizzi, working on that.

**Comment by arizzi** [comment]
same update as #18092 to avoid root files in release

**Comment by slava77** [comment]
@arizzi 
there are merge conflicts here.
Please rebase or remake&force-push

**Comment by arizzi** [comment]
everything is in the last commit, I'll cherry pick and force update

**Comment by arizzi** [comment]
can someone please trigger the tests?

**Comment by perrotta** [comment]
please test

**Comment by slava77** [comment]
@cmsbuild please test
to get cleaner comparisons now that CMSSW_9_1_X_2017-04-13-2300 is available (effective merge base for this PR)

File: PhysicsTools/PatAlgos/plugins/PATSecondaryVertexSlimmer.cc [hunk 1]
```diff
@@ -32,6 +37,7 @@ namespace pat {
 pat::PATSecondaryVertexSlimmer::PATSecondaryVertexSlimmer(const edm::ParameterSet& iConfig) :
   src_(consumes<reco::VertexCompositePtrCandidateCollection>(iConfig.getParameter<edm::InputTag>("src"))),
   srcLegacy_(mayConsume<std::vector<reco::Vertex> >(iConfig.getParameter<edm::InputTag>("src"))),
+  srcV0s_(mayConsume<reco::VertexCompositeCandidateCollection>(iConfig.getParameter<edm::InputTag>("src"))),
```
Comment (slava77) [inline PhysicsTools/PatAlgos/plugins/PATSecondaryVertexSlimmer.cc]: this runtime input type detection is getting only worse.
Please change the module configuration to pass the type to be used at configuration step.

@Dr15Jones is proliferation of this mayConsume to different types OK for the framework, or do we get some unwanted cost?

**Comment by slava77** [comment]
+1

for https://github.com/cms-sw/cmssw/pull/18300	4b691fa
- implemented as described
- jenkins tests pass and comparisons show differences in packed candidate delta plots where more cands have filled values (more so in the /lostTrack folder) as expected
- local test with PAT step running on 136.76 (2016E double muon) shows addition of slimmedKshortVertices and slimmedLambdaVertices modules which take under 0.1ms/event total and produce VertexCompositePtrCandidates collections with size consistent with what's in the PR description.

#18390 is created to possibly follow up on implementation of  PhysicsTools/PatAlgos/plugins/PATSecondaryVertexSlimmer.cc

File: PhysicsTools/PatAlgos/plugins/PATSecondaryVertexSlimmer.cc (hunk 1 above)
Comment (Dr15Jones) [inline PhysicsTools/PatAlgos/plugins/PATSecondaryVertexSlimmer.cc]: `mayConsumes` is presently (and may always) trigger prefetching of the item requested. It is possible to determine what type is available before calling `consumes`. This is done by registering a function to call when new products are registered. The registration is done by passing a functor to `callWhenNewProductsRegistered`. The documentation can be found here:
https://twiki.cern.ch/twiki/bin/view/CMSPublic/SWGuideEDMGetDataFromEvent#GetterOfProducts

**Pull request merged**